js/ui/windowManager.js

```diff
@@ -310,6 +311,149 @@ var EOSShellWobbly = new Lang.Class({
+                    }),
+                    stepper: new Animation.LinearStepper({ length: this.bridge.length })
+                }), detail.complete);
+                return true;
+                break;
+            case 'close':
+                this._startAnimation(new Animation.ZoomAnimation({
+                    from: new Animation.Box({
+                        top_left: new Animation.Vector({ x: detail.pointer[0], y: detail.pointer[1] }),
+                        bottom_right: new Animation.Vector({ x: detail.pointer[0], y: detail.pointer[1] })
+                    }),
+                    to: new Animation.Box({
+                        top_left: new Animation.Vector({ x: x, y: y }),
+                        bottom_right: new Animation.Vector({ x: x + width, y: y + height })
+                    }),
+                    stepper: new Animation.ReverseStepper({
+                        base_stepper: new Animation.LinearStepper({ length: this.bridge.length })
+                    })
+                }), detail.complete);
+                return true;
+                break;
+            case 'minimize':
+                this._startAnimation(new Animation.ZoomAnimation({
+                    from: new Animation.Box({
+                        top_left: new Animation.Vector({ x: detail.icon.x1, y: detail.icon.y1 }),
+                        bottom_right: new Animation.Vector({ x: detail.icon.x2, y: detail.icon.y2 })
+                    }),
+                    to: new Animation.Box({
+                        top_left: new Animation.Vector({ x: x, y: y }),
+                        bottom_right: new Animation.Vector({ x: x + width, y: y + height })
+                    }),
+                    stepper: new Animation.ReverseStepper({
+                        base_stepper: new Animation.LinearStepper({ length: this.bridge.length })
+                    })
+                }), detail.complete);
+                return true;
+                break;
+            case 'unminimize':
+                this._startAnimation(new Animation.ZoomAnimation({
+                    to: new Animation.Box({
+                        top_left: new Animation.Vector({ x: x, y: y }),
+                        bottom_right: new Animation.Vector({ x: x + width, y: y + height })
+                    }),
+                    from: new Animation.Box({
+                        top_left: new Animation.Vector({ x: detail.icon.x1, y: detail.icon.y1 }),
+                        bottom_right: new Animation.Vector({ x: detail.icon.x2, y: detail.icon.y2 })
+                    }),
+                    stepper: new Animation.LinearStepper({ length: this.bridge.length })
+                }), detail.complete);
+                return true;
+                break;
+            default:
+                return false;
+        }
+    },
+
+    remove: function() {
+        if (this.actor) {
+            this.actor.remove_effect(this);
+        }
```

Comment:
Nitpick No need for the brackets.

Reply:
Ah, I didn't get to this one + it might be too much rebasing for now. Feel free to fix it if you would like.